### ISSUE TITLE
Discover page counts and facets incorrect and confusing

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#832]
Following are issues identified during demo of the current discover interface at AGU:

 - [x] The numbers that appear next to filter facets in discover are sometimes incorrect.  Example, at present a search on the term "netcdf" results in a listing of 6 resources (not logged in) but the filter by resource type facet has netcdfResource 7.  
 - [x] Show facets even if there is one entry.  A search may return a long list of resources (too long to read).  The facets on the left show names for facets (like authors) where there are 2 or more.  But it would be helpful to know that there is one resource by an author matching a search and be able to find it.
 - [ ] Use consistent resource type names.  The term in the facet on the left does not match the term in the type column.  The resource type entry display names that appear on the drop down for select resource type during resource creation should be used consistently across the site.  Here netcdfResource is not an entry.  It should be "Multidimensional (NetCDF)".  Check others for consistency too.
 - [x] Colors and numbers do not match.  The netcdf search that results in 7 has a lighter color than other filters with smaller numbers.  
 - [x] Remove the color bar at the top.  I was asked what it means.  It is not obvious.  Lets jut let the color differences stand as they do and people may intuit the pattern eventually.  The color bar does not help clarify that hits are being ranked by color.

**Comment by hyi** [comment]
@dtarb @xuezhaokun @alvacouch @danames @horsburgh  How about if I add numbers and title to the color bar legend to make it something like this: 
![colorlegendexample](https://cloud.githubusercontent.com/assets/476302/11813297/a0e4df70-a30e-11e5-8d80-eed36e23d617.png)
Perhaps it is clearer to only use five color band like in the image above rather than eight as being used currently. I think adding colorbar legend is a way to go to make things clear, but agree the current colorbar legend being added in a rush needs more polishing to make it self-explanatory. If you think this proposal for polishing colorbar legend to make it self-explanatory like the one shown in the image, I can get this done pretty quickly, even for AGU if that is helpful and not too late.

**Comment by alvacouch** [comment]
I approve of Hong's suggestions. @xuezhaokun and I will discuss these issues tonight.

**Comment by xuezhaokun** [comment]
@alvacouch Sure. Talk to you later. I try to fix the above issue as much as possible now.

**Comment by danames** [comment]
I think the color bar is unnecessary on this page.

**Comment by xuezhaokun** [comment]
Following are issues identified during demo of the current discover interface at AGU:

 - [ ] The numbers that appear next to filter facets in discover are sometimes incorrect.  Example, at present a search on the term "netcdf" results in a listing of 6 resources (not logged in) but the filter by resource type facet has netcdfResource 7.  
 - [ ] Show facets even if there is one entry.  A search may return a long list of resources (too long to read).  The facets on the left show names for facets (like authors) where there are 2 or more.  But it would be helpful to know that there is one resource by an author matching a search and be able to find it.
 - [ ] Use consistent resource type names.  The term in the facet on the left does not match the term in the type column.  The resource type entry display names that appear on the drop down for select resource type during resource creation should be used consistently across the site.  Here netcdfResource is not an entry.  It should be "Multidimensional (NetCDF)".  Check others for consistency too.
 - [ ] Colors and numbers do not match.  The netcdf search that results in 7 has a lighter color than other filters with smaller numbers.  
 - [ ] Remove the color bar at the top.  I was asked what it means.  It is not obvious.  Lets jut let the color differences stand as they do and people may intuit the pattern eventually.  The color bar does not help clarify that hits are being ranked by color.

**Comment by xuezhaokun** [comment]
Following are issues identified during demo of the current discover interface at AGU:

 - [ ] The numbers that appear next to filter facets in discover are sometimes incorrect.  Example, at present a search on the term "netcdf" results in a listing of 6 resources (not logged in) but the filter by resource type facet has netcdfResource 7.  
This does not happen to me when I did test on my local. I am wondering whether it is better for us to do 'rebuild_index' instead of 'update_index' for updating our solr indexed resources 
 - [x] Show facets even if there is one entry.  A search may return a long list of resources (too long to read).  The facets on the left show names for facets (like authors) where there are 2 or more.  But it would be helpful to know that there is one resource by an author matching a search and be able to find it.
 - [x] Use consistent resource type names.  The term in the facet on the left does not match the term in the type column.  The resource type entry display names that appear on the drop down for select resource type during resource creation should be used consistently across the site.  Here netcdfResource is not an entry.  It should be "Multidimensional (NetCDF)".  Check others for consistency too.
 - [ ] Colors and numbers do not match.  The netcdf search that results in 7 has a lighter color than other filters with smaller numbers.  
 - [ ] Remove the color bar at the top.  I was asked what it means.  It is not obvious.  Lets jut let the color differences stand as they do and people may intuit the pattern eventually.  The color bar does not help clarify that hits are being ranked by color.

**Comment by xuezhaokun** [comment]
I am not sure why I cannot commit my changes on branch 832-improvements-on-discovery. I pushed the changes, but it does not show up on github.

**Comment by horsburgh** [comment]
I'm inclined to agree with @danames. Given that there are numbers in each of the bubbles, I don't think we need the color bar.

**Comment by hyi** [comment]
Agreed given that there are numbers in each of the bubbles, the colorbar legend is not a must-to-have (although I think it is a nice-to-have) and may not be necessary on this search page as of now. If the consensus is to remove the colorbar legend for now, @xuezhaokun, could you remove it as part of your feature branch for this issue? You just need to remove <svg> drawing for color bar legend in search.html and remove corresponding css styles used in hydroshare_core.css. Otherwise, if the consensus is to improve the colorbar legend as I suggested above, then @xuezhaokun can leave it out in his branch and I can work on it with a new branch.

**Comment by hyi** [comment]
@xuezhaokun Also, I think we can reverse the colormap to make the lighter color to represent smaller numbers and darker color to represent larger numbers, which is more in line with the colormap convention like the image I copied above. The lighter background color in the bubbles make the white numbers on the bubble hard to see, so we can change the font color for number labels on the bubbles to be darker, rather than the default white. Just some suggestions for your reference.

**Comment by dtarb** [comment]
I think that the consensus between @horsburgh @danames and me is to remove the color bar, but keep a color scale that goes from dark (many) to light (few).  Font and color adjustments suggested for readability are good.

**Comment by hyi** [comment]
@xuezhaokun Looks like the consensus is to remove the color bar on search page and reverse the current color map from light (few) to dark (many) and adjust number label font and color accordingly. It should be a simple change to incorporate into your current feature branch for this issue. Let me know if you want me to help with this ASAP, since I will take off for holiday break starting this Thursday.

**Comment by xuezhaokun** [comment]
@hyi  I can finish it by tonight. I created a branch 832-improvements-on-discovery. I am not sure why i cannot commit my changes. It seems the changes were committed successfully, but the changes are not shown on github repository. Any idea?

**Comment by hyi** [comment]
@xuezhaokun What does "git status" show? It may give you some clue on what is happening. Not sure what is going on since it never happened to me before.

**Comment by xuezhaokun** [comment]
@hyi it just said something like successfully pushed the changes. Just like the thing I got before when i commit and push my code. I will take a look at it. Thanks!

**Comment by hyi** [comment]
@xuezhaokun Make sure when you push, you add "origin your_branch_name" to git push command to make sure it pushes to the branch you want. Good luck in finding the root of the problem soon.

**Comment by mjstealey** [comment]
@xuezhaokun - when in doubt, git status is your friend. Also, if you're working from a different machine or sometimes even a different network than you normally work from you may need to tweak the configuration of your git environment.

In general, you should be following a pattern similar to this (using branch 832 as an example):
```bash
Do a bunch of work
...
$ git add FILE_1 <FILE_2 ... FILE_N>
$ git commit -m "[#832] your_super_descriptive_commit_message"
$ git push origin 832-improvements-on-discovery
```
You should be seeing some kind of output when you `git push`. 
Please post the terminal output if your problem persists.
Also, based on the output message we may need to look at configuring the remote, adjusting https or ssh, or other... hard to say without seeing the output at the terminal (assuming that you're using command line).

**Comment by dtarb** [comment]
@alvacouch @xuezhaokun I reviewed the discover interface on park.hydroshare.org.  It is vastly improved.  I checked the items above that are done. Here are a few additional suggestions for improvements, though none of these are show stoppers and I would be happy for this to be deployed and then these incrementally addressed.

- In the filter by resource type facet use descriptive resource type names, the same as what is used from the Create Resource Button Select resource type  under My Resources
![untitled](https://cloud.githubusercontent.com/assets/3456833/14371451/0d127ea6-fcf5-11e5-98c1-cabe4fbc3255.png)
- Allow sorting by title, type, First Author and Last modified by clicking on the header in the list similar to how My Resources works.
- Consider changing the "Clear All" button to "Show All".  While this clears all search restrictions, it shows all resources.  [Run this by @alvacouch first for his take on interface design though]
- Consider having the "Clear All" button also clear whatever words are in the search box.  [although this is also an interface design question that I'd like to defer to @alvacouch on]
- Consider combining Filter by Public and Filter By Discoverable into one facet named "Availability" with three options: Public, Discoverable, Private.  These I think are more concise than "true" and "false" currently presented.
- Sometimes clicking on the links in the search facets goes to a blank page with the word "discovery" on it.  This seems odd and I was not able to consistently repeat it.
- Couple map and Facet searching.  The map should be treated as just another facet.  A user should be able to drag a box on the map to identify a selection region (this may be possible now by shift drag, but the interface does not indicate this and does not show the area or resources selected).  With this active only the resources in (or touching) the selected region should be on the list.  This will necessitate an element to deactivate the map selection region selection.  Perhaps this could be done using a check box "Filter by map Selection".
